keras/utils/generic_utils.py

```diff
@@ -289,10 +289,16 @@ def update(self, current, values=None, force=False):
                 info += ' - %s:' % k
                 if isinstance(self.sum_values[k], list):
                     avg = self.sum_values[k][0] / max(1, self.sum_values[k][1])
-                    if abs(avg) > 1e-3:
-                        info += ' %.4f' % avg
+                    if isinstance(avg, float):
+                        if abs(avg) > 1e-3:
+                            info += ' %.4f' % avg
+                        else:
+                            info += ' %.4e' % avg
                     else:
-                        info += ' %.4e' % avg
+                        if abs(avg[-1]) > 1e-3:
```

Comment:
In this case, it is reasonable to use `mean` instead of `[-1]` (also below), because that's how losses and metrics are computed for multi-dimensional outputs.